### PR TITLE
contrib/database/sql: fix DBM service name when using peer.service

### PR DESCRIPTION
**Pull request by rarguelloF**
### What does this PR do?

Use the correct value for the DBM `dddbs` tag when using `peer.service`:
- If `peer.service` is present, or `peer.service` defaults calculation is enabled, use that as the value.
- Otherwise, fallback to the span service name.

### Motivation

Fix APM/DBM links when using `peer.service` for the service map.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Review by amarziali — APPROVED** [review]
the logic looks good

**Review by alexandre-normand — APPROVED** [review]
Thanks for the fix, @rarguelloF . This looks good to me. 

The only concern / improvement we could imagine is the one we discussed the other day where we'd have the `contrib/database/sql` package do the classic pattern of:
* start a span
* execute operation
* finish the span. 

This relates to this relatively [old issue](https://github.com/DataDog/dd-trace-go/pull/336) with the optional driver API/methods and we were curious if, now that those APIs are probably implemented in most drivers, we could get away with going back to that pattern. Of course, this change would need to happen in order for this fix to get simplified but I wanted to bring this up in case the Go tracer team was looking for more examples/motivation to refactor this code.

File: ddtrace/tracer/sqlcomment.go [hunk 1]
```diff
@@ -70,6 +71,38 @@ type SQLCommentCarrier struct {
...
+		return dbService
+	}
+	peerService, _ := c.getMeta(ext.PeerService)
+	if peerService != "" {
+		return peerService
+	}
+	if !tr.config.peerServiceDefaultsEnabled {
+		return dbService
+	}
+	if peerService, _ := getPeerServiceFromSource(c); peerService != "" {
```
Comment (dianashevchenko) [inline ddtrace/tracer/sqlcomment.go:102]: I ran go test with coverage, this function is only covered by tests lines 89 to 94. I suggest writing a small test, or adding more test cases in `TestSQLCommentCarrier`.

File: ddtrace/tracer/sqlcomment.go [hunk 2]
```diff
@@ -70,6 +71,38 @@ type SQLCommentCarrier struct {
...
+	if !ok {
+		return dbService
+	}
+	peerService, _ := c.getMeta(ext.PeerService)
+	if peerService != "" {
+		return peerService
+	}
+	if !tr.config.peerServiceDefaultsEnabled {
+		return dbService
+	}
```
Comment (dianashevchenko) [inline ddtrace/tracer/sqlcomment.go:101]: Suggestion of the code flow, if we are getting the tracer insance, we might as well see if the peerServiceDefaultsEnabled is 'true' right away. Thoughts?
```suggestion
	if tr, ok := internal.GetGlobalTracer().(*tracer); !ok || !tr.config.peerServiceDefaultsEnabled {
		return dbService
	}
	if peerService, _ := c.getMeta(ext.PeerService); peerService != "" {
		return peerService
	}
	
```

File: ddtrace/tracer/spancontext.go [hunk 3]
```diff
@@ -502,14 +503,22 @@ func setPeerService(s *span, cfg *config) {
 	}
 }
 
-// setPeerServiceFromSource sets peer.service from the sources determined
-// by the tags on the span. It returns the source tag name that it used for
+type spanMetaGetter interface {
```
Comment (dianashevchenko) [inline ddtrace/tracer/spancontext.go:506]: nit: I would suggest renaming the interface to metaProvider or smth like that

File: contrib/database/sql/propagation_test.go [hunk 4]
```diff
@@ -20,121 +24,164 @@ import (
...
-		prepared []string
-		executed []*regexp.Regexp
+		name           string
+		tracerOpts     []tracer.StartOption
+		registerOpts   []RegisterOption
+		opts           []Option
+		skipParentSpan bool
+		callDB         func(ctx context.Context, db *sql.DB) error
+		prepared       []string
+		executed       []*regexp.Regexp
```
Comment (dianashevchenko) [inline contrib/database/sql/propagation_test.go:55]: Can you please elaborate what are the changes in testing about? I see that they don't touch the code changed in ddtrace/tracer files

**Review by dianashevchenko — CHANGES_REQUESTED** [review]

**Pull request closed without merging**